src/plugins/credential_management/public/components/create_credential_wizard/components/header/add_form.tsx

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { Component } from 'react';
+import { FormattedMessage } from '@osd/i18n/react';
+
+import {
+  EuiForm,
+  EuiFormRow,
+  EuiFieldText,
+  EuiSelect,
+  EuiLink,
+  EuiButton,
+  EuiFieldPassword,
+  EuiText,
+} from '@elastic/eui';
+import { Credential } from '../../../../../../data_source/public';
+export class AddForm extends Component<any, any> {
+  state: any = {
+    showErrors: false,
+    nameShowErrors: false,
+    errors: [],
+    toasts: [],
+    credentialName: '',
+    description: '',
+    credentialMaterialsType: Credential.CredentialMaterialsType.UsernamePasswordType,
+    userName: '',
+    password: '',
+    dual: true,
+  };
+
+  createCredential = async () => {
```

Comment:
keeps functions/methods simpler (less than 75 lines).

Separate the validation & API call into 2 methods